File: common/src/main/java/com/astrion/common/packet/GamePacket.java
package com.astrion.common.packet;

/**
 * Wire format: [length(4 bytes)][packetType(1 byte)][json payload(N bytes)]
 */
public class GamePacket {
    private PacketType type;
    private String payload;

    public GamePacket() {}

    public GamePacket(PacketType type, String payload) {
        this.type = type;
        this.payload = payload;
    }

    public PacketType getType() { return type; }
    public void setType(PacketType type) { this.type = type; }
    public String getPayload() { return payload; }
    public void setPayload(String payload) { this.payload = payload; }
}

File: unity-client/Assets/Scripts/Network/GamePacket.cs
namespace Astrion.Network
{
    public class GamePacket
    {
        public PacketType Type { get; set; }
        public string Payload { get; set; }

        public GamePacket(PacketType type, string payload)
        {
            Type = type;
            Payload = payload;
        }
    }
}
